spring-cloud-generator/generate-one.sh

```diff
@@ -97,9 +97,10 @@ sed -i 's/{{parent-version}}/'"$parent_version"'/' "$starter_artifactid"/pom.xml
 # add module after line with pattern, check for existence. Also add readme line if $3 is set to 1.
 # args: 1 -  path-to-pom-file; 2 - string-pattern; 3 - 1 if need to generate readme line
 add_module_to_pom () {
-  xmllint --debug --nsclean --xpath  "//*[local-name()='module']/text()" $1 | sort | uniq | grep -q $starter_artifactid
+  xmllint --debug --nsclean --xpath  "//*[local-name()='module']/text()" $1 \
+    | sort | uniq | grep -q $starter_artifactid || module_list_is_empty=1
```

Comment:
@diegomarquezp  Could you help me better understand the reason/context behind needing this change with `module_list_is_empty`? 

When I run the existing script on a pom with empty module list, the `xmllint` command returns 1 (assigned to `found_library_in_pom`) and logic proceeds into the else block. Is there an edge case that I am missing here which this new logic resolves?

Reply:
Ah nvm I see the error now when trying to run the end-to-end workflow (was running the line separately earlier) - thank you for catching this!